botcraft/src/AI/Tasks/InventoryTasks.cpp

```diff
@@ -361,10 +361,12 @@ namespace Botcraft
                     {
                         Vector3<double> a_offset = neighbour_offsets[static_cast<int>(a)];
                         Vector3<double> b_offset = neighbour_offsets[static_cast<int>(b)];
-                        return player_orientation.dot(a_offset) < player_orientation.dot(b_offset); // a < b because a negative dot product means the vectors are in opposite directions IE the player is looking at the face.
+                        return player_orientation.dot(a_offset) > player_orientation.dot(b_offset); 
```

Comment:
It doesn't solve the issue I raised in the previous comment. Your return value is the same for each block in the world as it only depends on the player orentiation and the neighbour_offsets that are constants.

But for blocks at different relative positions with regard to the player, different faces should be returned (see the figure below).

![image](https://github.com/adepierre/Botcraft/assets/24371370/0ecd086e-625e-4744-9df7-41145413d41c)